risc0/zkvm/src/guest/env.rs

```diff
@@ -519,7 +704,8 @@ pub struct FdWriter<F: Fn(&[u8])> {
 }
 
 impl<F: Fn(&[u8])> FdWriter<F> {
-    fn new(fd: u32, hook: F) -> Self {
+    /// Creates a new FdWriter writing to the given file descriptor.
+    pub fn new(fd: u32, hook: F) -> Self {
```

Comment:
I thought of making this public so users could instantiate their own FdWriters if they want.

Reply:
I think we should have `env::write_fd` and `env::read_fd` to avoid having to make the `FdWriter` public.

Reply:
I think `FdWriter` has [always been public](https://github.com/risc0/risc0/commit/e16e860e303b2a750cfdff3ff00f83a1c431f2d0#diff-e1117fda42459c7cce9f894e70e03f3041b2405449de7be9cee9cc966a1db30dR337)?